scanner/src/main/kotlin/scanners/fossid/FossId.kt

```diff
@@ -221,17 +222,35 @@ class FossId(
                         .checkResponse("create project")
                 }
 
-                log.info { "Creating scan for URL $url and revision $revision ..." }
+                val scans = service.listScansForProject(user, apiKey, projectCode)
+                    .checkResponse("list scans for project").data
+                requireNotNull(scans)
```

Comment:
AFAIU this can be dropped as `withDataCheck = true` is the default for `checkResponse`.

Reply:
No it unfortunately does not work. Because the check is done in the `checkResponse` function, Kotlin cannot infer non-nullability from outside.

Reply:
In other code areas you expressed this via `data!!`, so I would have assumed that to be used here, too.

But a more speaking / semantically better way to express this is via `checkNotNull`: This function documents that the argument is guaranteed to be non-null. That's a different thing from `requireNotNull` which acknowledges that the value might be non-null, but a non-null value is required.

Reply:
Ok makes sense.
I try to prevent using `!!` as this is a code smell.